.github/workflows/PR-build.yml

```diff
@@ -0,0 +1,142 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT
+
+name: PR Build
+on:
+  pull_request:
+    branches: 
+      - main 
+    types: 
+      - synchronize
+      - reopened
+      - ready_for_review
+    paths-ignore:
+      - '**/*.md'
+      - '.github/**'
+      - '!.github/workflows/PR-build.yml'
+  workflow_dispatch:
```

Comment:
Why are we adding workflow_dispatch?

Reply:
In case we want to verify if the build with the main branch is whether success or not and if not success, what is the return error. Since the logs in Github Action will be [1-90 days for public repo ](https://docs.github.com/en/organizations/managing-organization-settings/configuring-the-retention-period-for-github-actions-artifacts-and-logs-in-your-organization)

Reply:
Why? We don't need this if we run build and test on merge to main.

Reply:
Would it better to always making sure the PR get up to dates with main branch though (e.g through [auto-update](https://github.com/marketplace/actions/auto-update), etc). In this way, we will always get the PR build on branch and avoid duplication build on main branch when merge

Reply:
I have added the options to "update branch" so developers can always merge branch when its out of date.
```
Whenever there are new changes available in the base branch, present an “update branch” option in the pull request.
```